src/deck/guides/3.0-upgrade.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Upgrade to Kong Gateway 3.x
+---
+
+Starting with {{site.base_gateway}} 3.0, paths using regular expressions must have a `~` prefix.
+The decK format version must be `3.0` to work with the new paths.
+
+If you [migrate your {{site.base_gateway}} database](/gateway/latest/upgrade/) from 2.8.x to 3.0, the paths in the database will have the `~` prefix added.
+This can cause configuration drift between state files and the database.
+
+Before using your state files to update the database, convert them into 3.0 format using `deck convert`.
```

Comment:
```suggestion
## Convert decK files

You can use `deck convert` to convert your decK files to 3.0: 

1. Run `deck-convert` against your 2.x state file:

    ```sh
    deck convert \
    --from kong-gateway-2.x \
    --to kong-gateway-3.x \
    --input-file kong2.8.yaml \
    --output-file new-kong3.0.yaml
    ```

    You can leave `input-file` out to convert all declarative configuration files in your working directory.


2. Validate the converted files in a test environment.

    These changes may not be fully correct or exhaustive, so perform manual audits of the generated file before applying
    the configuration in production.

```